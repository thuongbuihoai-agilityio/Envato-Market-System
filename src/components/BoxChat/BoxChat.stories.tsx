import { StoryObj, Meta } from '@storybook/react';

import BoxChat from '.';

const meta: Meta<typeof BoxChat> = {
  title: 'Custom Components/BoxChat',
  tags: ['autodocs'],
  component: BoxChat,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof BoxChat>;

export const Default: Story = {
  args: {},
};
