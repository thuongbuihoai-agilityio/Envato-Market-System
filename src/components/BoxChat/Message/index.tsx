import { Avatar, Box, Flex, Spacer, Text } from '@chakra-ui/react';

// Components
import VoiceChatMessage from './VoiceChatMessage';

// Constants
import { AVATAR_POSITION } from '@app/constants';

interface MessageProps {
  content?: string;
  avatar?: string;
  isImage?: boolean;
  avatarPosition?: AVATAR_POSITION;
  isOwnerMessage?: boolean;
  localeTime?: string;
}

const Message = ({
  content,
  avatar,
  avatarPosition = AVATAR_POSITION.BEFORE,
  isImage = false,
  isOwnerMessage = false,
  localeTime = new Date().toLocaleTimeString([], {
    hour: 'numeric',
    minute: '2-digit',
  }),
}: MessageProps) => (
  <Flex
    width="100%"
    justifyContent={
      avatarPosition === AVATAR_POSITION.AFTER ? 'flex-end' : 'flex-start'
    }
  >
    {avatarPosition === AVATAR_POSITION.BEFORE && (
      <Avatar src={avatar} borderColor="border.tertiary" w={9} h={9} mr={2} />
    )}

    <Flex
      align="flex-end"
      direction={isOwnerMessage ? 'row-reverse' : 'row'}
      mb="30px"
      alignItems="center"
    >
      <Box data-testid="image-container">
        {isImage ? (
          <VoiceChatMessage />
        ) : (
          <Text
            bg="background.component.secondary"
            p={3}
            ml={2}
            borderRadius={8}
            color="text.primary"
            data-testid="text-content"
          >
            {content}
          </Text>
        )}
      </Box>
      <Spacer />
      <Text
        ml={3}
        mr={3}
        fontSize="xs"
        color="text.textTime"
        fontWeight="medium"
        minW="max-content"
      >
        {localeTime}
      </Text>
    </Flex>

    {avatarPosition === AVATAR_POSITION.AFTER && (
      <Avatar
        src={avatar}
        borderColor="border.tertiary"
        w={9}
        h={9}
        ml={2}
        data-testid="avatar"
      />
    )}
  </Flex>
);

export default Message;
