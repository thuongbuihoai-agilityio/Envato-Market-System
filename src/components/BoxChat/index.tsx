import { memo } from 'react';
import { Box, Image, Heading, Flex, Input, Button } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Components
import Message from '@app/components/BoxChat/Message';

// Mocks
import { MESSAGE_TIME } from '@app/mocks';

const BoxChatComponent = (): JSX.Element => (
  <Box w="full" bg="background.body.quaternary" borderRadius="lg">
    <Flex
      direction="row"
      justifyContent="space-between"
      alignItems="center"
      justify="center"
      borderBottom="1px solid"
      borderBottomColor="border.tertiary"
      padding="24px 26px"
    >
      <Heading
        as="h3"
        fontWeight="semibold"
        color="text.primary"
        fontSize="2xl"
        textTransform="capitalize"
      >
        team chat
      </Heading>

      <Flex gap={5}>
        <Image
          src={IMAGES.USER_AVATAR.url}
          alt={IMAGES.USER_AVATAR.alt}
          w="52px"
          h={8}
        />

        <Button
          aria-label="btn-icon-plus"
          bg="secondary.800"
          w={9}
          h={9}
          borderRadius="50%"
        >
          <Image
            src="icons/plus.svg"
            alt={IMAGES.USER_AVATAR.alt}
            w={3.5}
            h={3.5}
          />
        </Button>
      </Flex>
    </Flex>

    <Box padding={{ base: '24px 20px', lg: '38px 35px' }}>
      <Message
        content="Hi, What can i help you with?"
        avatarPosition="before"
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={MESSAGE_TIME}
      />

      <Message
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        isImage
        localeTime={MESSAGE_TIME}
      />

      <Flex direction="row-reverse">
        <Message
          content="Hello, I want to know more about your product"
          avatarPosition="after"
          isOwnerMessage
          avatar={IMAGES.CHAT_USER_AVATAR.url}
          localeTime={MESSAGE_TIME}
        />
      </Flex>

      <Message
        content="Sure, I can help you with that"
        avatarPosition="before"
        avatar={IMAGES.CHAT_USER_AVATAR.url}
        localeTime={MESSAGE_TIME}
      />

      <Flex justify="center" align="center">
        <Box
          border="1px solid"
          borderColor="border.secondary"
          p="0 20px"
          mt={5}
          borderRadius="lg"
          w="full"
        >
          <Flex direction="row" alignItems="center">
            <Image
              src={IMAGES.ATTACH.url}
              alt={IMAGES.ATTACH.alt}
              cursor="pointer"
            />

            <Input
              variant="authentication"
              ml={5}
              type="number"
              _dark={{
                border: 'none',
              }}
              sx={{ border: 'none', padding: 0 }}
              placeholder="Type your message here"
            />

            <Image
              src={IMAGES.MICRO.url}
              alt={IMAGES.MICRO.alt}
              cursor="pointer"
            />
          </Flex>
        </Box>
        <Image
          mt={5}
          ml={5}
          src={IMAGES.SEND.url}
          alt={IMAGES.SEND.alt}
          cursor="pointer"
        />
      </Flex>
    </Box>
  </Box>
);

const BoxChat = memo(BoxChatComponent);

export default BoxChat;
