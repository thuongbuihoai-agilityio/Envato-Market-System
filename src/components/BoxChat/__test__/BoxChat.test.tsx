import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import BoxChat from '@app/components/BoxChat';

test('BoxChat component renders correctly', () => {
  const { container } = render(<BoxChat />);
  expect(container).toMatchSnapshot();
});
