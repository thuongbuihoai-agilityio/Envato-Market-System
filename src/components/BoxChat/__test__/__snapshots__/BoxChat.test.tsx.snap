// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxChat component renders correctly 1`] = `
<div>
  <div
    class="css-1tid5r0"
  >
    <div
      class="css-1mdtjo1"
    >
      <h3
        class="chakra-heading css-gh62sw"
      >
        team chat
      </h3>
      <div
        class="css-1j52cd2"
      >
        <img
          alt="This is the user avatar"
          class="chakra-image__placeholder css-jqgv83"
          src="/icons/user.svg"
        />
        <button
          aria-label="btn-icon-plus"
          class="chakra-button css-18tuhly"
          type="button"
        >
          <img
            alt="This is the user avatar"
            class="chakra-image__placeholder css-k1r8rc"
            src="/icons/user.svg"
          />
        </button>
      </div>
    </div>
    <div
      class="css-gpjsqf"
    >
      <div
        class="css-k008qs"
      >
        <span
          class="chakra-avatar css-gyntuh"
        >
          <svg
            aria-label=" avatar"
            class="chakra-avatar__svg css-16ite8i"
            role="img"
            viewBox="0 0 128 128"
          >
            <path
              d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
              fill="currentColor"
            />
            <path
              d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="css-1kz9q0"
        >
          <div
            class="css-0"
            data-testid="image-container"
          >
            <p
              class="chakra-text css-rgei1j"
              data-testid="text-content"
            >
              Hi, What can i help you with?
            </p>
          </div>
          <div
            class="css-17xejub"
          />
          <p
            class="chakra-text css-1qzadu6"
          >
            3:51 PM
          </p>
        </div>
      </div>
      <div
        class="css-k008qs"
      >
        <span
          class="chakra-avatar css-gyntuh"
        >
          <svg
            aria-label=" avatar"
            class="chakra-avatar__svg css-16ite8i"
            role="img"
            viewBox="0 0 128 128"
          >
            <path
              d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
              fill="currentColor"
            />
            <path
              d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="css-1kz9q0"
        >
          <div
            class="css-0"
            data-testid="image-container"
          >
            <img
              alt="Message Image"
              class="chakra-image css-14gt3bc"
            />
          </div>
          <div
            class="css-17xejub"
          />
          <p
            class="chakra-text css-1qzadu6"
          >
            3:51 PM
          </p>
        </div>
      </div>
      <div
        class="css-tz0gqg"
      >
        <div
          class="css-k008qs"
        >
          <div
            class="css-53hlci"
          >
            <div
              class="css-0"
              data-testid="image-container"
            >
              <p
                class="chakra-text css-rgei1j"
                data-testid="text-content"
              >
                Hello, I want to know more about your product
              </p>
            </div>
            <div
              class="css-17xejub"
            />
            <p
              class="chakra-text css-1qzadu6"
            >
              3:51 PM
            </p>
          </div>
          <span
            class="chakra-avatar css-ntspms"
            data-testid="avatar"
          >
            <svg
              aria-label=" avatar"
              class="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="css-k008qs"
      >
        <span
          class="chakra-avatar css-gyntuh"
        >
          <svg
            aria-label=" avatar"
            class="chakra-avatar__svg css-16ite8i"
            role="img"
            viewBox="0 0 128 128"
          >
            <path
              d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
              fill="currentColor"
            />
            <path
              d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="css-1kz9q0"
        >
          <div
            class="css-0"
            data-testid="image-container"
          >
            <p
              class="chakra-text css-rgei1j"
              data-testid="text-content"
            >
              Sure, I can help you with that
            </p>
          </div>
          <div
            class="css-17xejub"
          />
          <p
            class="chakra-text css-1qzadu6"
          >
            3:51 PM
          </p>
        </div>
      </div>
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-dyz3ep"
        >
          <div
            class="css-u4p24i"
          >
            <img
              alt="Attach icon"
              class="chakra-image css-4g6ai3"
              src="/icons/attach.svg"
            />
            <input
              class="chakra-input css-3qiee2"
              placeholder="Type your message here"
              type="number"
            />
            <img
              alt="Micro icon"
              class="chakra-image css-4g6ai3"
              src="/icons/micro.svg"
            />
          </div>
        </div>
        <img
          alt="Send icon"
          class="chakra-image css-qgs394"
          src="/icons/send.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
